File: prolog/locations/toilet.pl
:- multifile die/0, ate/1, talked/1.

% ============== Info ==============

info(toilet) :-
  write('It''s the toilet you recognize as your own. It''s not the first'), nl,
  write('nor the last time you are here. From behind the doors you hear a'), nl,
  write('muffled, sore voice.'), nl.

% ============= Objects =============

% Paper roll
at(paper_roll, toilet).
do_describe(paper_roll) :-
  write('Toilet paper, usually used to clean yourself'), nl,
  !.
do_eat(paper_roll) :-
  write('Tasty paper! Good job, you feel fulfilled. Also slightly nauseous.'), nl,
  !.

% ============= People =============

% Voice from toilet
person_at(voice_from_toilet, toilet).
do_talk(voice_from_toilet) :-
  ate(toilet_paper),
  write('Did I just hear you eating paper out there?'), nl,
  write('Bart are you feeling alright? Please, leave me alone.'), nl,
  !.
do_talk(voice_from_toilet) :-
  talked(voice_from_toilet),
  write('Toilet paper, PLEASE!'), nl,
  !.
do_talk(voice_from_toilet) :-
  write('Bart, thank god you are here, I need toilet paper NOW!'), nl,
  write('Please bring me a roll and I will reward you kindly.'), nl,
  !.
do_give(flyer, voice_from_toilet) :-
  write('Oh, that. Yea yesterday''s party was a blast.'), nl,
  write('I am glad that at least you''re safe. Haven''t heard from neither Rad'), nl,
  write('or Mat since yesterday, hope they''re safe.'), nl,
  !.
do_give(toilet_paper, voice_from_toilet) :-
  write('Oh, thank you so much!'), nl,
  write('Here, take this coin, you can do whatever with it.'), nl,
  assert(in_inventory(old_five_zloty)),
  !.

% ============= Possible goto Locations =============

path(your_room, toilet).

File: haskell/Locations/Toilet.hs
module Locations.Toilet where

import Data.Map (Map)
import qualified Data.Map as Map

import Datatypes

-- ============= Objects ============

toilet_paper = ItemMap{itemName = "Toilet Paper"
                   , itemDescription = ["It\'s a toilet paper"]
                   , itemEat = ["You\'ve eaten the toilet paper. It was not a really good idea."]
                   , itemEatDeath = ["You\'ve been eating the toilet paper and suddenly you choked and suffocated to death. Good job!"]
                   , itemEatDeathChance = 10 
                   }

-- ============= People =============

voice_from_toilet = PersonMap {personName = "Voice from toilet"
                          , personTalk = ["Bart, thank god you are here, I need toilet paper NOW!", 
                                          "Please bring me a roll and I will reward you kindly."]
                          , personTalkAgain = ["Toilet paper, PLEASE!"]
                          , personTalkWithItem = Map.fromList [(
                              "toilet_paper", ["Thank you, Bart, you are a true friend!"]
                          )]
                          , personGive = Map.fromList [
                              ("toilet_paper", ["Thank you"])
                            ]
                          , personItems = []}

-- ============= Location ===========

toilet = LocationMap {locationDescription = ["It\'s the toilet you recognize as your own. It\'s not the first",
                                             "nor the last time you are here. From behind the doors you hear a",
                                             "muffled, sore voice."]
                            , locationItems = Map.fromList [
                              ("toilet_paper", toilet_paper)
                              ]
                            , locationPeople = Map.fromList [
                                ("voice_from_toilet", voice_from_toilet)
                              ]
                            , locationPaths = ["your_room"]}